Fix force_save_fsdp_all_gather and split_fsdp_prefetch (#283)

sanketpurandare opened the pull request:
Stacked PRs:
 * #290
 * __->__#283


--- --- ---

### Fix force_save_fsdp_all_gather and split_fsdp_prefetch (#283)


* Fix force_save_fsdp_all_gather and split_fsdp_prefetch

wconstab reviewed (approved):
i partly checked but it looks like the new helper functions are just refactoring not logic change from the old code.

nit: PR desc had incomplete sentence 

"We verify if one the users of this last node is used in backward. I"

sanketpurandare committed:
Fix force_save_fsdp_all_gather and split_fsdp_prefetch (#283)

* Fix force_save_fsdp_all_gather and split_fsdp_prefetch

stack-info: PR: https://github.com/meta-pytorch/autoparallel/pull/283, branch: sanketpurandare/stack/1
